cheatsheets/gleam-for-javascript-users.md

```diff
@@ -0,0 +1,700 @@
+let size = 1
+io.debug(size) // prints 1
+print_it() // prints 50
+```
+
+### Variables type annotations
+
+#### JavaScript/TypeScript
+
+In plain JavaScript there are no static types, but in TypeScript variables can
+optionally be annotated with types.
+
+```typescript
+const some_list: number[] = [1, 2, 3];
+```
+
+#### Gleam
+
+In Gleam type annotations can optionally be given when binding variables.
+
+```gleam
+let some_list: List(Int) = [1, 2, 3]
+```
+
+Gleam will check the type annotation to ensure that it matches the type of the
+assigned value. It does not need annotations to type-check your code, but you
+may find it useful to annotate variables to hint to the compiler that you want a
+specific type to be inferred.
+
+## Functions
+
+#### JavaScript
+
+In JavaScript, you can define functions with the `function` keyword, or assign
+anonymous functions to variables.
+
+```javascript
+function sum(x, y) {
+  return x + y;
+}
+
+const mul = (x, y) => x * y;
+```
+
+#### Gleam
+
+Gleam's functions are declared using a syntax similar to Rust. Gleam's anonymous
+functions have a similar syntax, just without the function name.
+
+```gleam
+pub fn sum(x, y) {
+  x + y
+}
+
+let mul = fn(x, y) { x * y }
+mul(1, 2)
+```
+
+### Function capturing with `_`
+
+To turn a multi-parameter function into a function with just one parameter,
+Gleam has a special function-capturing syntax:
+
+```gleam
+fn add(a: Int, b: Int) -> Int {
+  a + b
+}
+
+let add2 = fn(x) { add(x, 2) }
+// is equivalent to:
+let add2 = add(_, 2)
+```
+
+This is particularly useful in combination with the [pipe operator](#piping).
+
+### Exporting functions
+
+#### JavaScript
+
+In both JavaScript and Gleam, functions are private by default. In JavaScript,
+functions can be made public with the `export` keyword.
+
+```javascript
+// this is public
+export function sum(x, y) {
+  return x + y;
+}
+
+// this is private
+function mul(x, y) {
+  return x * y;
+}
+```
+
+#### Gleam
+
+In Gleam functions need the `pub` keyword to be public.
+
+```gleam
+// this is public
+pub fn sum(x, y) {
+  x + y
+}
+
+// this is private
+fn mul(x, y) {
+  x * y
+}
+```
+
+### Function type annotations
+
+#### JavaScript/TypeScript
+
+In TypeScript, you can annotate the types of function parameters and return
+values.
+
+```typescript
+function sum(x: number, y: number): number {
+  return x + y;
+}
+```
+
+The choice of whether, and exactly how to annotate a function can influence how
+type-safe your code is.
+
+#### Gleam
+
+In Gleam, functions can **optionally** have their argument and return types
+annotated. These type annotations will always be checked by the compiler and
+throw a compilation error if not valid. The compiler will still type check your
+program using type inference if annotations are omitted.
+
+```gleam
+pub fn add(x: Int, y: Int) -> Int {
+  x + y
+}
+
+pub fn mul(x: Int, y: Int) -> Bool { // compile error, type mismatch
+  x * y
+}
+```
+
+### Referencing functions
```

Comment:
I think we can remove this section as JS programmers wouldn't expect anything else